File: src/main/java/edu/cornell/cs/apl/viaduct/imp/ast/types/BooleanType.java
package edu.cornell.cs.apl.viaduct.imp.ast.types;

public final class BooleanType implements ImpBaseType {
  private static final BooleanType INSTANCE = new BooleanType();

  private BooleanType() {}

  public static BooleanType create() {
    return INSTANCE;
  }

  @Override
  public String toString() {
    return "bool";
  }
}

File: src/main/kotlin/edu/cornell/cs/apl/viaduct/syntax/types/BooleanType.kt
package edu.cornell.cs.apl.viaduct.syntax.types

import edu.cornell.cs.apl.prettyprinting.Document
import edu.cornell.cs.apl.prettyprinting.styled
import edu.cornell.cs.apl.viaduct.syntax.values.BooleanValue
import edu.cornell.cs.apl.viaduct.syntax.values.Value

/** The type of booleans. */
object BooleanType : ValueType {
    override val defaultValue: Value
        get() = BooleanValue(false)

    override val asDocument: Document =
        Document("bool").styled(ValueTypeStyle)
}
